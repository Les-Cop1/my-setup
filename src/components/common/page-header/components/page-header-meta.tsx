import React from 'react'

import { PageHeaderMetaProps } from '@components'
import { classNames } from '@helpers'

export const PageHeaderMeta: React.FC<PageHeaderMetaProps> = ({
  icon: Icon,
  label,
  backgroundColor = 'white',
  ...props
}) => {
  let iconClassName: string
  let labelClassName: string
  switch (backgroundColor) {
    case 'dark':
      iconClassName = 'text-zinc-200'
      labelClassName = 'text-zinc-200'
      break
    case 'white':
    case 'light':
    default:
      iconClassName = 'text-slate-400'
      labelClassName = 'text-slate-500'
  }

  return (
    <div className={classNames('mt-2 flex items-center text-sm', labelClassName)} {...props}>
      {Icon && <Icon className={classNames('mr-1.5 h-5 w-5 flex-shrink-0', iconClassName)} aria-hidden="true" />}
      {label}
    </div>
  )
}
