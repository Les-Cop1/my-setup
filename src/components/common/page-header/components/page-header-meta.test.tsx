import React from 'react'

import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

import { PageHeaderMeta } from './page-header-meta'

describe('Page Header Meta', () => {
  test('PageHeaderMeta should have correct background color', () => {
    render(<PageHeaderMeta label="default" testid="default" />)
    render(<PageHeaderMeta label="light" testid="light" backgroundColor="light" />)
    render(<PageHeaderMeta label="white" testid="white" backgroundColor="white" />)
    render(<PageHeaderMeta label="dark" testid="dark" backgroundColor="dark" />)
    expect(screen.getByTestId('default')).toHaveClass('text-slate-500')
    expect(screen.getByTestId('light')).toHaveClass('text-slate-500')
    expect(screen.getByTestId('white')).toHaveClass('text-slate-500')
    expect(screen.getByTestId('dark')).toHaveClass('text-zinc-200')
  })

  test('PageHeaderMeta should have label', () => {
    render(<PageHeaderMeta label="label" testid="default" />)
    expect(screen.getByText('label')).toBeInTheDocument()
  })

  test('Snapshot', () => {
    const { container } = render(<PageHeaderMeta label="Hello" />)
    expect(container).toMatchSnapshot()
  })
})
