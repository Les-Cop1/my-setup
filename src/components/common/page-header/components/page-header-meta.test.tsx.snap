// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header Meta Snapshot 1`] = `
<div>
  <div
    class="mt-2 flex items-center text-sm text-slate-500"
  >
    Hello
  </div>
</div>
`;
